### ISSUE TITLE
Add examples with other frameworks

### ISSUE THREAD

**Issue by ericchiang** [coreos/dex#320]
This was brought up on the google group. It would be really good to have working examples of using dex with other authorization frameworks. At the very least, it's good to make sure dex works with other OpenID Connect client implementations.

Suggested frameworks:
- [ ] Node.js [passport-openidconnect](https://www.npmjs.com/package/passport-openidconnect)
- [ ] Python [django-oidc-provider](https://github.com/juanifioren/django-oidc-provider)

**Comment by davidyum** [comment]
I was able to get a node express app + passport-openidconnect working with Dex by modifying the passport module slightly in my client app.
 
I captured the request from the passport-openidconnect and the go client app and they look slightly different. I believe Dex expects authorization header (encoded clientid and secret) as part of the request to the "/token" endpoint:
![image](https://cloud.githubusercontent.com/assets/1501075/13821921/1506312a-eb7a-11e5-9018-633bab1ff5a0.png)
The passport request wasn't including the authorization header as part of the request.

**Comment by ericchiang** [comment]
Interesting. This may be a bug on our end. Do you know how the credentials are begin sent by passport-openidconnect?

**Comment by davidyum** [comment]
Passport-openidconnect sample Req:
https://gist.github.com/davidyum/d73609dc7125cc23a6f8

Go Client sample Req:
https://gist.github.com/davidyum/693cbf70b53dc2ec1122

**Comment by ericchiang** [comment]
Sorry what is this a dump of? The dex logs?

```
(EXTRA *http.Request=&{POST /token HTTP/1.1 1 1 map[User-Agent:[Node-oauth] Content-Length:[286] 
Connection:[close] Content-Type:[application/x-www-form-urlencoded]] 0xc820523b80 286 [] true localhost:5556 
map[grant_type:[authorization_code] redirect_uri:[http://127.0.0.1:3000/callback] client_id:[rASLY1bNGjfthEe5FOEw-0ds9ZJAItMDYTSl_mnqe_Q=@127.0.0.1] client_secret:[hUdVk1Pz-A97FcfQ9W6OopfFWOgmU2lMfCcjV3Mi6GUvjZPXB5CzxsgYYbokFyCysrDi3KfmT4jQWkqePbaSAdE0PXmdCmF- ] code:[ARAbQ3dc4Fw=]] <nil> map[] 127.0.0.1:59367 /token <nil> <nil>})
```

It looks like the auth might be in the body from `Content-Type: application/x-www-form-urlencoded`

**Comment by davidyum** [comment]
Yes, I placed logs in dex to capture the two requests.
It seems like Dex is decoding the Authorization header from the request and setting the username as the client id and setting the password as the client secret.
This is in the server/http.go file handleTokenFunc() .

**Comment by ericchiang** [comment]
It looks like the oauth2 does allow you to pass creds via the request body.

>    Alternatively, the authorization server MAY support including the
   client credentials in the request-body using the following
   parameters:

>   client_id
         REQUIRED.  The client identifier issued to the client during
         the registration process described by Section 2.2.

>   client_secret
         REQUIRED.  The client secret.  The client MAY omit the
         parameter if the client secret is an empty string.

>   __Including the client credentials in the request-body using the two
   parameters is NOT RECOMMENDED and SHOULD be limited to clients unable
   to directly utilize the HTTP Basic authentication scheme__ (or other
   password-based HTTP authentication schemes).  The parameters can only
   be transmitted in the request-body and MUST NOT be included in the
   request URI.

https://tools.ietf.org/html/rfc6749#section-2.3 (emphasis my own)

**Comment by ericchiang** [comment]
Yeah, the oauth package that the passport-openidconnect does this https://github.com/ciaranj/node-oauth/blob/ec6a1b2eafd2206f248d83e5ddccf911f68cb3fe/lib/oauth2.js#L164-L176

**Comment by davidyum** [comment]
Thanks for looking into it Eric. It seems like an open issue for the node-oauth package: 
https://github.com/ciaranj/node-oauth/issues/175

Adding the line below to the node-oauth oauth2.js line 155-159 fixed the issue:
```
  var post_data= querystring.stringify( params );
  var post_headers= {
       'Content-Type': 'application/x-www-form-urlencoded',
       'Authorization': 'Basic ' + new Buffer(this._clientId + ':' + this._clientSecret).toString('base64')
   };
```

**Comment by ericchiang** [comment]
Maybe dependent on https://github.com/coreos/dex/issues/376?

**Comment by Fmstrat** [comment]
Has anyone been able to get Passport.js working with Dex? I am unable to still.
